api/src/test/java/io/opentelemetry/trace/propagation/HttpTraceContextTest.java

```diff
@@ -166,6 +166,18 @@ void extract_SampledContext() {
                 TRACE_ID_BASE16, SPAN_ID_BASE16, SAMPLED_TRACE_OPTIONS, TRACE_STATE_DEFAULT));
   }
 
+  @Test
+  void extract_NullCarrier() {
+    Map<String, String> carrier = new LinkedHashMap<>();
+    carrier.put(TRACE_PARENT, TRACEPARENT_HEADER_SAMPLED);
+    assertThat(
+            getSpanContext(
+                httpTraceContext.extract(Context.current(), null, (c, k) -> carrier.get(k))))
```

Comment:
This test here demonstrates one case where a null carrier might be sensibly used.

Reply:
*sensibly is relative here but at least it demonstrates how checking for null could be surprising